cmd/channel.go

```diff
@@ -45,81 +48,141 @@ func channelCmdChain() *cobra.Command {
 		},
 	}
 
-	channelCfgCmd := &cobra.Command{
-		Use:   "cfg",
-		Short: "return the configuration active channel",
+	channelGenCmd := &cobra.Command{
+		Use:   "gen",
+		Short: "generate a mazzaroth channel",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Generate Key
+			pub, priv, err := crypto.GenerateEd25519KeyPair()
+			if err != nil {
+				return err
+			}
+			fmt.Println("channel address:", crypto.ToHex(pub))
+			fmt.Println("channel private key:", crypto.ToHex(priv))
+			fmt.Println("channel Id:", crypto.ToHex(pub))
+			// TODO
+			// self signed cert for channel
+			// mazzaroth.io cert generate for channel
+			return nil
+		},
+	}
+
+	channelDeleteCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "delete a channel contract",
 		RunE: func(cmd *cobra.Command, args []string) error {
 
-			client, err := mazzaroth.NewMazzarothClient(mazzaroth.WithAddress(viper.GetString(address)))
+			sender, err := xdr.IDFromHexString(viper.GetString(publicKey))
 			if err != nil {
 				return err
 			}
 
-			config, err := client.ChannelConfig(cmd.Context(), viper.GetString(channelId))
+			channelId, err := xdr.IDFromHexString(viper.GetString(channelId))
 			if err != nil {
 				return err
 			}
 
-			v, err := json.MarshalIndent(config, "", "\t")
+			pk, err := crypto.FromHex(viper.GetString(privateKey))
 			if err != nil {
 				return err
 			}
 
-			fmt.Println(string(v))
-			return nil
-		},
-	}
+			tx, err := mazzaroth.Transaction(sender, channelId).Contract(mazzaroth.GenerateNonce(), defaultBlockExpirationNumber).Delete().Sign(pk)
+			if err != nil {
+				return err
+			}
 
-	channelGenCmd := &cobra.Command{
-		Use:   "gen",
-		Short: "generate a mazzaroth channel",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			// Generate Key
-			_, pub, err := crypto.GenerateEd25519KeyPair()
+			client, err := mazzaroth.NewMazzarothClient(mazzaroth.WithAddress(viper.GetString(channelAddress)))
 			if err != nil {
 				return err
 			}
-			fmt.Println("channel address:", crypto.ToHex(pub))
-			// TODO
-			// self signed cert for channel
-			// mazzaroth.io cert generate for channel
+
+			id, _, err := client.TransactionSubmit(cmd.Context(), tx)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println("transaction id:", id)
```

Comment:
The print for Pause and Delete should convert the id to hex like the Transaction Submit output:

```
txId := hex.EncodeToString(id[:])
```

Reply:
pushed